File: dataset/Vanilla/OPENAI_gpt-5-2025-08-07/Task72.java
import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.nio.charset.StandardCharsets;
import java.util.Arrays;
import java.util.Base64;

public class Task72 {
    // WARNING: Using a static IV is insecure and only for demonstration/testing.
    private static final byte[] STATIC_IV = "0123456789ABCDEF".getBytes(StandardCharsets.UTF_8);

    public static byte[] encrypt(byte[] key, byte[] data) throws Exception {
        int blockSize = 16;
        byte[] padded = padPKCS7(data, blockSize);

        SecretKeySpec keySpec = new SecretKeySpec(key, "AES");
        IvParameterSpec ivSpec = new IvParameterSpec(STATIC_IV);

        Cipher cipher = Cipher.getInstance("AES/CBC/NoPadding");
        cipher.init(Cipher.ENCRYPT_MODE, keySpec, ivSpec);

        return cipher.doFinal(padded);
    }

    private static byte[] padPKCS7(byte[] data, int blockSize) {
        int padding = blockSize - (data.length % blockSize);
        if (padding == 0) padding = blockSize;
        byte[] out = Arrays.copyOf(data, data.length + padding);
        Arrays.fill(out, data.length, out.length, (byte) padding);
        return out;
    }

    private static String toHex(byte[] bytes) {
        StringBuilder sb = new StringBuilder(bytes.length * 2);
        for (byte b : bytes) sb.append(String.format("%02x", b));
        return sb.toString();
    }

    public static void main(String[] args) throws Exception {
        byte[] key = "ThisIsA16ByteKey".getBytes(StandardCharsets.UTF_8); // 16-byte key (AES-128)

        String[] tests = new String[] {
            "",
            "Hello, AES-CBC!",
            "The quick brown fox jumps over the lazy dog",
            "1234567890ABCDEF1234",
            "Data with \u0000 nulls \u0000 inside"
        };

        for (int i = 0; i < tests.length; i++) {
            byte[] plaintext = tests[i].getBytes(StandardCharsets.UTF_8);
            byte[] ciphertext = encrypt(key, plaintext);
            String b64 = Base64.getEncoder().encodeToString(ciphertext);
            System.out.println("Test " + (i + 1) + " Base64: " + b64);
            System.out.println("Test " + (i + 1) + " Hex:    " + toHex(ciphertext));
        }
    }
}

File: dataset/Vanilla/OPENAI_gpt-5-2025-08-07/Task72.py
# WARNING: Using a static IV is insecure and only for demonstration/testing.
from Crypto.Cipher import AES
import base64

STATIC_IV = b"0123456789ABCDEF"

def pad_pkcs7(data: bytes, block_size: int = 16) -> bytes:
    padding = block_size - (len(data) % block_size)
    if padding == 0:
        padding = block_size
    return data + bytes([padding]) * padding

def encrypt(key: bytes, data: bytes) -> bytes:
    cipher = AES.new(key, AES.MODE_CBC, STATIC_IV)
    padded = pad_pkcs7(data, 16)
    return cipher.encrypt(padded)

def _to_hex(b: bytes) -> str:
    return ''.join(f"{x:02x}" for x in b)

if __name__ == "__main__":
    key = b"ThisIsA16ByteKey"  # 16-byte key (AES-128)
    tests = [
        b"",
        b"Hello, AES-CBC!",
        b"The quick brown fox jumps over the lazy dog",
        b"1234567890ABCDEF1234",
        b"Data with \x00 nulls \x00 inside",
    ]
    for i, t in enumerate(tests, 1):
        ct = encrypt(key, t)
        print(f"Test {i} Base64: {base64.b64encode(ct).decode('utf-8')}")
        print(f"Test {i} Hex:    {_to_hex(ct)}")